### ISSUE TITLE
Raise sqlstate with `message` does not return `message` (used with custom http status codes).

### ISSUE THREAD

**Issue by robertsosinski** [PostgREST/postgrest#2176]
* PostgreSQL version: 13.1
* PostgREST version: 9.0.0
* Operating system: Docker on Win 10

When I raise an error with `sqlstate` I specify a `message`:

```
RAISE sqlstate 'PT402' using
  message = 'Payment Required',
  detail = 'Quota exceeded',
  hint = 'Upgrade your plan';
```

However, the `message` does not come along with the response:

```
{
    "hint":"Upgrade your plan",
    "details":"Quota exceeded"
}
```

If I raise with a built in error, the message comes through though:

```
RAISE  insufficient_privilege using 
  detail = 'Quota exceeded',
  hint = 'Upgrade your plan';
```

You can see here:

```
{
    "hint": "Upgrade your plan",
    "details": "Quota exceeded",
    "code": "42501",
    "message": "insufficient_privilege"
}
```

Would be great to get the `message` specified in the `raise` statement when specifying custom http status codes.

Thanks!

**Comment by steve-chavez** [comment]
> RAISE sqlstate 'PT402' using
>   message = 'Payment Required',
>   detail = 'Quota exceeded',
>   hint = 'Upgrade your plan';

IIRC the `message` there is passed as the status message. So if you do:

```sql
 RAISE sqlstate 'PT452' using
   message = 'My Special Status',
   detail = 'Quota exceeded',
   hint = 'Upgrade your plan';
```

It would respond with:

```
HTTP/1.1 452 My Special Status 
```

I thought this could be used for defining new status codes/messages but perhaps the most useful behavior is to auto-generate the status messages for known status codes(similar to [response.status](https://postgrest.org/en/stable/api.html#setting-response-status-code)) and leave them blank for unknown status codes.

---

Relevant code:

https://github.com/PostgREST/postgrest/blob/dcf7ade5bc70a31fe0e083e1527b74125a94aeda/src/PostgREST/Error.hs#L172-L182

**Comment by robertsosinski** [comment]
That makes sense, I just took a look and indeed the `message` field is visible with the HTTP code. I think this works, but if we could also get the same `message` and `code` in the JSON response body that built in errors get, that would be more consistent.  The response would look like:

plpgsql
```
 RAISE sqlstate 'PT452' using
   message = 'My Special Status',
   detail = 'Quota exceeded',
   hint = 'Upgrade your plan';
```

response
```
HTTP/1.1 452 My Special Status 

{
    "hint":"Upgrade your plan",
    "details":"Quota exceeded",
    "code": "PT452",
    "message": "My Special Status"
}
```

This would now make customer http errors have the same fields/format as built in errors.


I'm using the `message` field in the response body for an interface. Because custom messages do not have it, I have to now also check the response object to try and fetch it from the status, which would require some refactoring on how I've abstracted interaction with my postgrest endpoint. As such, I'll probably just keep doing what I'm doing, and when there is no `message` field in the response, just put "error".

Adding the status code would also make it easier to configure error flows too.

Interested in your thoughts.

**Comment by steve-chavez** [comment]
I see no problem in including the code/message in the body, IIRC it was done that way to avoid having redundant info; and as you mention, it'd better for consistency in client libraries.

**Comment by steve-chavez** [comment]
Btw, on https://github.com/PostgREST/postgrest/pull/1926 we were trying to normalize the error messages: having them all respond with a "code" field.

Some postgrest errors don't include the `details` and `hint` field too. WDYT it'd be better? To have them always shown as `details: null, hint: null` or continue keeping them as "undefined"?

**Comment by laurenceisla** [comment]
Fixed in #1926
